lib/pure/collections/critbits.nim

```diff
@@ -179,11 +179,7 @@ proc `[]=`*[T](c: var CritBitTree[T], key: string, val: T) =
 template get[T](c: CritBitTree[T], key: string): T =
   let n = rawGet(c, key)
   if n == nil:
-    when compiles($key):
```

Comment:
I like the other version much better than relying on the new "fact" that "everything has a dollar operator" (which isn't true in your PR either btw, callback types cannot be stringified).

Reply:
Valid point with the callback type. As I said, this `everything can be printed` should be an RFC and I will provide it. Still changes like this were important to detect cases non printable types and find solutions for them.